plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -321,299 +264,285 @@ def rpm_dependency_conversion(solvable, unit_name, unit_flags, unit_evr,
+            artifact_name = artifact_name.encode('utf-8')
+
+        if artifact_arch is not None:
+            artifact_arch = artifact_arch.encode('utf-8')
+
+        artifact_evr = libsolv_formatted_evr(artifact_epoch, artifact_version, artifact_release)
+
+        module_artifacts_conversion(pool, solvable, artifact_name, artifact_evr, artifact_arch)
+
+    pool.createwhatprovides()  # TODO: It would be great to do this less often
+    module_dependencies_conversion(pool, solvable, unit.get('dependencies', []))
+
     return solvable
 
 
-def unit_solvable_converter_factory(*attribute_factories):
-    return lambda solv_repo, unit: unit_solvable_converter(solv_repo, unit, *attribute_factories)
-
-
-rpm_unit_solvable_factory = unit_solvable_converter_factory(
-    # An RPM content unit nests dependencies in a dict format
-    # Would be provided by pulp_rpm
-    plain_attribute_factory('name'),
-    evr_attribute,
-    plain_attribute_factory('arch'),
-    plain_attribute_factory('vendor'),
-    rpm_dependency_attribute_factory('requires'),
-    rpm_dependency_attribute_factory('provides'),
-    rpm_dependency_attribute_factory('recommends'),
-    rpm_filelist_conversion,
-    rpm_basic_deps,
-)
-
-
-@utf8_conversion
-@multiattr_conversion(
-    attr_conversion('name')(),
-    attr_conversion('stream')(),
-    attr_conversion('version')(),
-    attr_conversion('context')()
-)
-def module_solvable_name(solvable, name, stream, version, context):
-    """
-    Create a solvable name from module attributes: module:<name>:<stream>:<version>:<context>
+def module_dependencies_conversion(pool, module_solvable, dependency_list):
     """
-    return 'module:{name}:{stream}:{version!s}:{context}'.format(
-        name=name,
-        stream=stream,
-        version=version,
-        context=context,
-    )
+    Process the module dependency list.
 
+    So for example for following input:
+        dependency_list = [{'gtk': ['1'], 'foo': ['1']}]
 
-module_name_attribute = setattr_conversion('name')(module_solvable_name)
+    The resulting solv.Dep expression will be:
+        ((module(gtk) with module(gtk:1)) and (module(foo) with module(foo:1)))
 
+    This Dep expression is then applied to the REQUIRES: deparray of the module solvable.
 
-@multiattr_conversion(
-    module_solvable_name,
-    utf8_conversion(attr_conversion('name')()),
-    utf8_conversion(attr_conversion('stream')()),
-    attr_conversion('version')(),
-    utf8_conversion(attr_conversion('arch', default='noarch')())
-)
-def module_basic_deps(solvable, solvable_name, name, stream, version, arch):
-    """
-    Create the basic module `Provides:` and `Obsoletes:` relations
+    :param pool: The libsolv pool that owns the module
+    :type pool: solv.Pool
+    :param module_solvable: A solvable representing the module
+    :type module_solvable: solv.Solvable
+    :param dependency_list: List of dictionaries representing modulemd dependency data.
+    :type dependency_list: List
```

Comment:
It probably should be a lowercase `list`.

Reply:
+1